test/fork/migration-v2.js

```diff
@@ -545,24 +566,31 @@ describe('v2 migration', function () {
       this.governance,
     );
 
-    this.claimsReward = newClaimsReward;
-    this.pool = pool;
-
+    this.memberRoles = await ethers.getContractAt('MemberRoles', this.memberRoles.address);
     this.mcr = await ethers.getContractAt('MCR', mcr.address);
     this.cover = await ethers.getContractAt('Cover', coverProxyAddress);
-    this.coverMigrator = await ethers.getContractAt('CoverMigrator', coverMigrator.address);
-    this.memberRoles = await ethers.getContractAt('MemberRoles', this.memberRoles.address);
-
     const tokenControllerAddress = await this.master.contractAddresses(toUtf8Bytes('TC'));
     this.tokenController = await ethers.getContractAt('TokenController', tokenControllerAddress);
-
     const pooledStakingAddress = await this.master.contractAddresses(toUtf8Bytes('PS'));
     this.pooledStaking = await ethers.getContractAt('LegacyPooledStaking', pooledStakingAddress);
-
+    this.pool = pool;
+    this.coverMigrator = await ethers.getContractAt('CoverMigrator', coverMigrator.address);
     const gatewayAddress = await this.master.contractAddresses(toUtf8Bytes('GW'));
     this.gateway = await ethers.getContractAt('LegacyGateway', gatewayAddress);
+
+    // TODO check if this dependency makes sense
```

Comment:
this is fixed in `feature/migrate-pooledstaking` as discussed.

Upgrading CR first or better yet making the NXMToken dependency of TokenController immutable solves the problem.

When upgrading CR this code path is triggered:

https://github.com/NexusMutual/smart-contracts/blob/2faaf7602bd61eeead8db8dd921835baabd6f636/contracts/modules/governance/NXMaster.sol#L149


If TokenController is upgraded in the same batch prior to CR, it does NOT have its dependency registry filled up, because the call to `changeDependentContractAddress` happens after all contracts are upgraded, in a loop for all upgrades.